src/mapping/PartitionOfUnityMapping.hpp

```diff
@@ -0,0 +1,385 @@
+#pragma once
+
+#include <Eigen/Core>
+#include <numeric>
+
+#include "com/Communication.hpp"
+#include "io/ExportVTU.hpp"
+#include "mapping/CreatePartitioning.hpp"
+#include "mapping/Partition.hpp"
+#include "mapping/RadialBasisFctBaseMapping.hpp"
+#include "mesh/Filter.hpp"
+#include "precice/types.hpp"
+#include "query/Index.hpp"
+#include "utils/Event.hpp"
+#include "utils/IntraComm.hpp"
+
+namespace precice {
+extern bool syncMode;
+
+namespace mapping {
+
+/**
+ * @brief Mapping using partition of unity decomposition strategies
+ */
+template <typename RADIAL_BASIS_FUNCTION_T>
+class PartitionOfUnityMapping : public Mapping {
+public:
+  /**
+   * @brief Constructor.
+   *
+   * @param[in] constraint Specifies mapping to be consistent or conservative.
+   * @param[in] dimension Dimensionality of the meshes
+   * @param[in] parameter shape parameter or support radius of the interpolation RBF
+   * @param[in] verticesPerPartition Estimate for the number of vertices to be clustered together
+   * @param[in] relativeOverlap Overlap between partitions, where 1 would correspnd to a complete overlap, 0 to distance of 2 x radius between partitions
+   */
+  PartitionOfUnityMapping(
+      Mapping::Constraint constraint,
+      int                 dimension,
+      double              parameter,
+      Polynomial          polynomial,
+      unsigned int        verticesPerPartition,
+      double              relativeOverlap,
+      bool                projectToInput);
+
+  /// Computes the mapping coefficients from the in- and output mesh.
+  virtual void computeMapping() override;
+
+  /// Removes a computed mapping.
+  virtual void clear() override;
+
+  /// tag the vertices required for the mapping
+  virtual void tagMeshFirstRound() override;
+
+  /// nothing to do here
+  virtual void tagMeshSecondRound() override;
+
+private:
+  precice::logging::Logger _log{"mapping::PartitionOfUnityMapping"};
+
+  std::vector<Partition<RADIAL_BASIS_FUNCTION_T>> _partitions;
+
+  // Shape parameter or support radius for the RBF interpolant,
+  // only required for the Partition instantiation
+  const double _parameter;
+
+  // Input parameter
+  const unsigned int _verticesPerPartition;
+  const double       _relativeOverlap;
+  const bool         _projectToInput;
+
+  // Derived parameter
+  double averagePartitionRadius = 0;
+
+  /// true if the mapping along some axis should be ignored
+  /// has currently only dim x false entries, as integrated polynomials are irrelevant
+  std::vector<bool> _deadAxis;
+
+  Polynomial _polynomial;
+
+  // Holds the output vertex -> partition association. Outer vector has the size of the output mesh and inner vector size of the associated partitions
+  std::vector<std::vector<VertexID>> _partMap;
```

Comment:
Would a `map` be easier to understand?
```suggestion
  // Holds the partition -> output vertex association. Outer vector has the size of the output mesh and inner vector size of the associated partitions
  std::vector<std::vector<VertexID>> _partMap;
```

Reply:
I was able to get rid of this data structure